### PR TITLE
Snowbridge on Westend

### PR DESCRIPTION
**Pull request by yrong**
### Context

Since Rococo is now deprecated, we need another testnet to detect bleeding-edge changes to Substrate, Polkadot, & BEEFY consensus protocols that could brick the bridge.

It's the mirror PR of https://github.com/Snowfork/polkadot-sdk/pull/157 which has reviewed by Snowbridge team internally.

Synced with @acatangiu about that in channel https://matrix.to/#/!gxqZwOyvhLstCgPJHO:matrix.parity.io/$N0CvTfDSl3cOQLEJeZBh-wlKJUXx7EDHAuNN5HuYHY4?via=matrix.parity.io&via=parity.io&via=matrix.org

### CODE REVIEW

**Comment by claravanstaden** [comment]
@yrong you need to add a prdoc for this PR, some examples here: https://github.com/paritytech/polkadot-sdk/tree/master/prdoc It's a simple markdown file (should be called pr_5074.prdoc in this case), describing the change. :)

**Review by claravanstaden — APPROVED** [review]
Nice! 🥳

**Review by serban300 — APPROVED** [review]

File: cumulus/parachains/runtimes/bridge-hubs/bridge-hub-westend/src/xcm_config.rs [hunk 1]
```diff
@@ -261,3 +279,22 @@ impl cumulus_pallet_xcm::Config for Runtime {
...
+pub struct XcmFeeManagerFromComponentsBridgeHub<WaivedLocations, HandleFee>(
+	PhantomData<(WaivedLocations, HandleFee)>,
+);
+impl<WaivedLocations: Contains<Location>, FeeHandler: HandleFee> FeeManager
+	for XcmFeeManagerFromComponentsBridgeHub<WaivedLocations, FeeHandler>
+{
+	fn is_waived(origin: Option<&Location>, fee_reason: FeeReason) -> bool {
+		let Some(loc) = origin else { return false };
+		if let Export { network, destination: Here } = fee_reason {
+			return !(network == EthereumNetwork::get())
```
Comment (acatangiu) [inline cumulus/parachains/runtimes/bridge-hubs/bridge-hub-westend/src/xcm_config.rs:292]: Don't short-circuit when export network is not Ethereum.

```suggestion
			if network == EthereumNetwork::get() {
				return false
			}
```

**Review by acatangiu — APPROVED** [review]

File: cumulus/parachains/runtimes/bridge-hubs/bridge-hub-westend/src/xcm_config.rs (hunk 1 above)
Comment (yrong) [reply cumulus/parachains/runtimes/bridge-hubs/bridge-hub-westend/src/xcm_config.rs]: https://github.com/paritytech/polkadot-sdk/pull/5074/commits/909c40cfdfad2f552fa14a53bb83e1a2384d2f73

File: cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/lib.rs [hunk 2]
```diff
@@ -145,6 +145,7 @@ pub type SignedExtra = (
 		bridge_to_bulletin_config::OnBridgeHubRococoRefundRococoBulletinMessages,
 	),
 	cumulus_primitives_storage_weight_reclaim::StorageWeightReclaim<Runtime>,
+	frame_metadata_hash_extension::CheckMetadataHash<Runtime>,
```
Comment (serban300) [inline cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/lib.rs:148]: @yrong this breaks the bridge zombienet tests. I can either remove it here or add it to the bridge relayer in order to address this issue. So just wanted to ask if you actually need it here in order to understand how to fix this. The same for bridge hub westend.

Also I think the runtime version should be incremented 

Related to: https://github.com/paritytech/polkadot-sdk/issues/5296

**Pull request merged**